File: backend/src/main/java/com/tripfriend/domain/review/dto/CommentResponseDto.java
package com.tripfriend.domain.review.dto;

import com.tripfriend.domain.review.entity.Comment;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
public class CommentResponseDto {
    private Long commentId;
    private String content;
    private Long reviewId;
    private Long memberId;
    private String memberName;
    private String profileImage;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public CommentResponseDto(Comment comment, String memberName) {
        this.commentId = comment.getCommentId();
        this.content = comment.getContent();
        this.reviewId = comment.getReview().getReviewId();
        this.memberId = comment.getMember().getId();
        this.memberName = memberName;
        this.profileImage = null;
        this.createdAt = comment.getCreatedAt();
        this.updatedAt = comment.getUpdatedAt();
    }

    public CommentResponseDto(Comment comment, String memberName, String profileImage) {
        this(comment, memberName);
        this.profileImage = profileImage;
    }
}

File: backend/src/main/java/com/tripfriend/domain/review/dto/CommentResponseDto.kt
package com.tripfriend.domain.review.dto

import com.tripfriend.domain.review.entity.Comment
import java.time.LocalDateTime

data class CommentResponseDto(
    val commentId: Long? = null,
    val content: String = "",
    val reviewId: Long? = null,
    val memberId: Long? = null,
    val memberName: String = "",
    var profileImage: String? = null,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null
) {
    constructor(comment: Comment, memberName: String) : this(
        commentId = comment.commentId,
        content = comment.content,
        reviewId = comment.review?.reviewId,
        memberId = comment.member?.id,
        memberName = memberName,
        profileImage = null,
        createdAt = comment.createdAt,
        updatedAt = comment.updatedAt
    )

    constructor(comment: Comment, memberName: String, profileImage: String?) : this(
        commentId = comment.commentId,
        content = comment.content,
        reviewId = comment.review?.reviewId,
        memberId = comment.member?.id,
        memberName = memberName,
        profileImage = profileImage,
        createdAt = comment.createdAt,
        updatedAt = comment.updatedAt
    )
}